test/ipfs-factory/daemon-spawner.js

```diff
@@ -66,30 +56,34 @@ function spawnEphemeralNode (callback) {
     if (err) {
       return callback(err)
     }
+
     // Note: we have to set each config value
     // independently since the config/replace endpoint
     // doesn't work as expected
     series([
       (cb) => {
+        node.getConfig((err, config) => {
+          if (err) throw err
+          console.log(config)
+          cb()
+        })
+      },
+      (cb) => {
+        console.log('HERE')
         const configValues = {
-          Bootstrap: [],
-          Discovery: {},
-          'HTTPHeaders.Access-Control-Allow-Origin': ['*'],
-          'HTTPHeaders.Access-Control-Allow-Credentials': 'true',
-          'HTTPHeaders.Access-Control-Allow-Methods': ['PUT', 'POST', 'GET']
+          'Bootstrap': [],
+          'Discovery': {},
+          'API.HTTPHeaders.Access-Control-Allow-Origin': ['*'],
+          'API.HTTPHeaders.Access-Control-Allow-Credentials': 'true',
+          'API.HTTPHeaders.Access-Control-Allow-Methods': ['PUT', 'POST', 'GET']
         }
 
         eachSeries(Object.keys(configValues), (configKey, cb) => {
-          node.setConfig(`API.${configKey}`, JSON.stringify(configValues[configKey]), cb)
+          const configVal = JSON.stringify(configValues[configKey])
+          node.setConfig(configKey, configVal, cb)
         }, cb)
       },
       (cb) => node.startDaemon(cb)
-    ], (err) => {
-      if (err) {
-        return callback(err)
-      }
-
-      callback(null, node)
-    })
+    ], (err) => callback(err, node))
```

Comment:
@dignifiedquire could you take a look at this function, I'm getting errors setting the config as the config is always empty, not sure what I messed up right now.